File: go/observer/VideoChannel.go
package main

import "fmt"

type VideoChannel struct {
	subscribers []Subscriber
}

func (vc *VideoChannel) Subscribe(sub Subscriber) {
	vc.subscribers = append(vc.subscribers, sub)
}

func (vc *VideoChannel) Unsubscribe(sub Subscriber) {
	for i, s := range vc.subscribers {
		if s == sub {
			vc.subscribers = append(vc.subscribers[:i], vc.subscribers[i+1:]...)
			break
		}
	}
}

func (vc *VideoChannel) UploadVideo(videoTitle string) {
	fmt.Printf("Video \"%s\" uploaded.\n", videoTitle)
	vc.notifySubscribers(videoTitle)
}

func (vc *VideoChannel) notifySubscribers(videoTitle string) {
	for _, sub := range vc.subscribers {
		sub.Update(videoTitle)
	}
}

File: python/observer/VideoChannel.py
from typing import List
from Subscriber import Subscriber

class VideoChannel:
    def __init__(self):
        self._subscribers: List[Subscriber] = []

    def subscribe(self, subscriber: Subscriber) -> None:
        self._subscribers.append(subscriber)

    def unsubscribe(self, subscriber: Subscriber) -> None:
        self._subscribers = [
            s for s in self._subscribers if s != subscriber
        ]

    def upload_video(self, video_title: str) -> None:
        print(f'Video "{video_title}" uploaded.')
        self._notify_subscribers(video_title)

    def _notify_subscribers(self, video_title: str) -> None:
        for subscriber in self._subscribers:
            subscriber.update(video_title)
